### PR TITLE
How password hashing occurs

### LINKED ISSUE #3554: How password hashing occurs
**Introduction**
`Password hashing` is the process of converting a raw password into a series of characters using a hashing algorithm. This process of hashing can be referred to as `SHA-1(Secure Hashing Algorithm)`. It can be further modified to be more secure by adding random data into the series of characters through a process known as `salting`. This improves the security of the password by making it more complex for a third party to decode the password hash.

In this article, we will learn through processes that occur from when you input your password to how it is converted to a hash. We will further see how the password is decoded back to a raw format by the computer. With that, we will learn how to protect our password so as it is now easily decoded by unauthorized parties.

**Key Takeaways**
In this article, we will learn:
1. different types of encryption
2. different formats in password encryption such as SHA-1, MD5, AES, etc
3. Accessing your password hash in both Windows and Kali Linux OS.
4. Protecting your password from a brute-force attack and unauthorized personnel.

**Reference**
N/A

### PR DESCRIPTION
**Pull request by Koros-Meshack**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3554

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/how-password-hashing-occurs/index.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+For this case will use the `SHA-256 hashing algorithm` and `python`. To make it simpler, we will use [replit.com website](https://replit.com/languages/python3). Just like any IDE,`replit` allows you to write, run and save code. For our case, we will run a code on the platform and see its output. Be Keen to note every line of code since it explains the steps that take place during hashing.
+
+
+First, navigate to the website [replit.com](https://replit.com/languages/python3). 
+
+Let's import the sha256 constructor from the `hashlib module` by running the command below.
+
+```
+    from hashlib import sha256
+```
```
Comment (Ericgacoki) [inline content/articles/how-password-hashing-occurs/index.md:27]: Please add the language identifier.

Comment (Koros-Meshack) [reply content/articles/how-password-hashing-occurs/index.md]: changes made @Ericgacoki

**Comment by ahmadmardeni1** [comment]
plagiarism check done ✅

I think this article is light on substance. What do you think? @Ericgacoki

**Comment by Ericgacoki** [comment]
Greetings @ahmadmardeni1, I think the author has covered quite enough content on hashing algorithms and hash tables. I also believe that there's a need to add more content to this article to make it stand out.

@Koros-Meshack, you can apply the existing knowledge to come up with a sample project that will help the readers understand this topic better.

**Comment by Koros-Meshack** [comment]
Thanks for highlighting that @Ericgacoki
I'll make the necessary updates to add value to the article and add more content as you have mentioned
I'll ping you as soon as am done
cc: @ahmadmardeni1

**Comment by Koros-Meshack** [comment]
I have made necessary changes and Content Addition @Ericgacoki

**Review by Ericgacoki — CHANGES_REQUESTED** [review]
Hello @Koros-Meshack, Nice tutorial -very easy to follow along. I have one suggestion attached below...

File: content/articles/how-password-hashing-occurs/index.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+```bash
+example321
+```
+
+**Output**
+
+```bash
+f84ad825bc3e5ccc7f5e35b6a5e83cb79a998f106e79410ae382c7d43019b6d6
+```
```
[the remark this answers is not in the document]
Comment (Koros-Meshack) [reply content/articles/how-password-hashing-occurs/index.md:123]: All changes made @Ericgacoki 
Please continue with the review

**Review by Ericgacoki — APPROVED** [review]

**Comment by WanjaMIKE** [comment]
Merged #3718 into master

**Pull request merged**